compiler/rustc_middle/src/ty/layout.rs

```diff
@@ -2473,15 +2473,9 @@ where
                     }
                 }
 
-                // dyn*
+                // dyn* (both fields are usize-sized)
                 ty::Dynamic(_, _, TraitObjectRepresentation::Sized) => {
-                    TyMaybeWithLayout::TyAndLayout(
-                        tcx.layout_of(
```

Comment:
Note that this fn is to get the layout of one field, so it doesn't make sense to return the whole layout here. Also, no need to call layout_of here either, it gets called later on I think.

Reply:
Ah, this probably explains why I was getting infinite recursion with some changes I was trying to make the other day...